### DISCUSSION TITLE
How do I get this working?

### DISCUSSION THREAD

**Discussion by BeckPhillips** [null2264/CobbleGen#79] [Q&A] [answered]
Right, I've gotten the config file working, and I'm trying to use this mod to make an Andesite generator that works with basalt blocks. I've gone in to the pre-existing stone variant generator and removed all the stuff for Diorite, cobblestone, and granite, then changed the bedrock to basalt.

And now, the generator is just not working at all. All it does is make cobblestone.
What's going on?

`"stoneGen": {
			"minecraft:basalt": [
				{
					"id": "minecraft:andesite",
					"weight": 100.0,
				},
			],
		},`

**Comment by null2264** [comment]
Are you trying to make cobblestone generator or stone generator?

Stone gens is lava flowing to a water from **the top**, while cobble gens is lava flowing to a water from **the side**. Since you said "All it does is make cobblestone", I assume you were trying to make cobble gen? If so, replace "stoneGen" with "cobbleGen"

**Reply by BeckPhillips** [reply]
I was trying to make a variation of a basalt generator, at first - Though I will admit I misunderstood how they worked at the time of posting this, even when making them normally, I still couldn't get any of my replacements to be recognized. The only thing that's worked is changing the cobblestone generator that also generates ore to make Andesite instead of cobblestone, but it doesn't seem to work if I remove the ores, so now the generator gives me a ton of excess ores ;-;

**Reply by null2264** [reply]
Is the config u sent on initial post is the whole thing or just snipet of it? I'm assuming u're only sending just a snippet of ur config file. If not, the config must start and ends with `{}`.

If you don't mind, can you attach the config file instead? so that I have more context...

**Reply by BeckPhillips** [reply]
Yeah, it's just a small section. Here's the full file:

{
	"formatVersion": "1.0",
	"cobbleGen": [
		{
			"id": "minecraft:cobblestone",
			"weight": 100.0,
			"minY": 0,
		},
		{
			"id": "minecraft:cobbled_deepslate",
			"weight": 100.0,
			"maxY": 0,
		},
	],
	"stoneGen": [
		{
			"id": "minecraft:stone",
			"weight": 100.0,
		},
	],
	"basaltGen": [
		{
			"id": "minecraft:basalt",
			"weight": 100.0,
		},
	],
	"customGen": {
		"cobbleGen": {
			"minecraft:basalt": [
				{
					"id": "minecraft:emerald_ore",
					"weight": 1.0,
				},
				{
					"id": "minecraft:diamond_ore",
					"weight": 1.0,
				},
				{
					"id": "minecraft:diorite",
					"weight": 1.0,
				},
				{
					"id": "minecraft:gold_ore",
					"weight": 2.0,
				},
				{
					"id": "minecraft:iron_ore",
					"weight": 3.0,
				},
				{
					"id": "minecraft:coal_ore",
					"weight": 3.0,
				},
				{
					"id": "minecraft:andesite",
					"weight": 90.0,
				},
			],
		},
		"stoneGen": {
			"minecraft:basalt": [
				{
					"id": "minecraft:andesite",
					"weight": 100.0,
				},
			],
		},
		"basaltGen": {
			"minecraft:bedrock": [
				{
					"id": "minecraft:end_stone",
					"weight": 100.0,
					"dimensions": [
						"minecraft:the_end",
					],
				},
				{
					"id": "minecraft:blackstone",
					"weight": 100.0,
					"excludedDimensions": [
						"minecraft:overworld",
					],
				},
			],
		},
	},
}

**Reply by null2264** [reply]
Uhh.. seems to works just fine for me...

customgen->cobblegen
![image](https://github.com/user-attachments/assets/af65bb16-cdd1-4e34-ac75-f50bb1bbcb05)

customgen->stonegen
![image](https://github.com/user-attachments/assets/ce9ef36b-8627-4a6f-8a73-df15a0e462c2)

**Reply by BeckPhillips** [reply, marked as answer]
I ended up changing what I was attempting, and don't have the original intent anymore - but I was trying to get a generator that would only make Andesite.
I've got something that works now, though, so it's honestly fine.

**Reply by null2264** [reply]
Oh, alright then :+1:
